src/header/common/link.rs

```diff
@@ -0,0 +1,1141 @@
+
+        let expected_link = Link::new(vec![link_value]);
+
+        let link = Header::parse_header(&vec![link_header.to_vec()].into());
+        assert_eq!(link.ok(), Some(expected_link));
+    }
+
+    #[test]
+    fn test_link_multiple_link_headers() {
+        let first_link = LinkValue::new("/TheBook/chapter2")
+            .push_rel(RelationType::Previous)
+            .set_title_star("UTF-8'de'letztes%20Kapitel");
+
+        let second_link = LinkValue::new("/TheBook/chapter4")
+            .push_rel(RelationType::Next)
+            .set_title_star("UTF-8'de'n%c3%a4chstes%20Kapitel");
+
+        let third_link = LinkValue::new("http://example.com/TheBook/chapter2")
+            .push_rel(RelationType::Previous)
+            .push_rev(RelationType::Next)
+            .set_title("previous chapter");
+
+        let expected_link = Link::new(vec![first_link, second_link, third_link]);
+
+        let raw = MemBuf::from(b"GET /super_short_uri/and_whatever HTTP/1.1\r\nHost: \
+                                  hyper.rs\r\nAccept: a lot of things\r\nAccept-Charset: \
+                                  utf8\r\nAccept-Encoding: *\r\nLink: </TheBook/chapter2>; \
+                                  rel=\"previous\"; title*=UTF-8'de'letztes%20Kapitel, \
+                                  </TheBook/chapter4>; rel=\"next\"; title*=\
+                                  UTF-8'de'n%c3%a4chstes%20Kapitel\r\n\
+                                  Access-Control-Allow-Credentials: None\r\nLink: \
+                                  <http://example.com/TheBook/chapter2>; \
+                                  rel=\"previous\"; rev=next; title=\"previous chapter\"\
+                                  \r\n\r\n".to_vec());
+
+        let (mut res, _) = ServerTransaction::parse(&raw).unwrap().unwrap();
+
+        let link = res.headers.remove::<Link>().unwrap();
+
+        assert_eq!(link, expected_link);
+    }
+
+    #[test]
+    fn test_link_display() {
+        let link_value = LinkValue::new("http://example.com/TheBook/chapter2")
+            .push_rel(RelationType::Previous)
+            .set_anchor("/anchor/example/")
+            .push_rev(RelationType::Next)
+            .push_href_lang(langtag!(de))
+            .push_media_desc(MediaDesc::Screen)
+            .set_title("previous chapter")
+            .set_title_star("title* unparsed")
+            .set_media_type(Mime(Text, Plain, vec![]))
+            .set_link_extension("link-extension unparsed");
+
+        let link = Link::new(vec![link_value]);
+
+        let mut link_header = String::new();
+        write!(&mut link_header, "{}", link).unwrap();
+
+        let expected_link_header = "<http://example.com/TheBook/chapter2>; \
+            rel=\"previous\"; anchor=\"/anchor/example/\"; \
+            rev=\"next\"; hreflang=de; media=\"screen\"; \
+            title=\"previous chapter\"; title*=title* unparsed; \
+            type=\"text/plain\"; link-extension=link-extension unparsed";
```

Comment:
This doesn't seem right. If I'm reading the ABNF from https://tools.ietf.org/html/rfc5988#section-5 correctly

> ```
>   Link           = "Link" ":" #link-value
>   link-value     = "<" URI-Reference ">" *( ";" link-param )
>   link-param     = ( ( "rel" "=" relation-types )
>                  | ( "anchor" "=" <"> URI-Reference <"> )
>                  | ( "rev" "=" relation-types )
>                  | ( "hreflang" "=" Language-Tag )
>                  | ( "media" "=" ( MediaDesc | ( <"> MediaDesc <"> ) ) )
>                  | ( "title" "=" quoted-string )
>                  | ( "title*" "=" ext-value )
>                  | ( "type" "=" ( media-type | quoted-mt ) )
>                  | ( link-extension ) )
>   link-extension = ( parmname [ "=" ( ptoken | quoted-string ) ] )
>                  | ( ext-name-star "=" ext-value )
> ```

`link-extension` isn't a literal but either a single attribute as a `parmname` (defined in https://tools.ietf.org/html/rfc5987#section-3.2.1) or a key value pair separated by `=` where the key is a `parmname` and the value is either a `ptoken` or a `quoted-string` (defined in https://tools.ietf.org/html/rfc2616#section-2.2).

With that in mind, my understanding is that headers like

```
Link: <https://example.com>; rel="previous"; foo="bar"; baz="qux" 
```

are valid. `foo` and `bar` are both `link-extension`s.